--- roundcubemail-0.2.1/index.php~	2009-03-16 20:19:56.519373134 +0200
+++ roundcubemail-0.2.1/index.php	2009-03-16 20:19:47.749882178 +0200
@@ -122,9 +118,12 @@
     $OUTPUT->show_message('sessionerror', 'error');
     $RCMAIL->kill_session();
   }
+} 
+// faq page
+else if ($RCMAIL->task=='faq') {
+	$OUTPUT->send($RCMAIL->task);
 }
 
-
 // check client X-header to verify request origin
 if ($OUTPUT->ajax_call) {
   if (!$RCMAIL->config->get('devel_mode') && !rc_request_header('X-RoundCube-Referer')) {
--- roundcubemail-0.2.1/program/include/rcmail.php~	2009-03-16 20:19:56.536247276 +0200
+++ roundcubemail-0.2.1/program/include/rcmail.php	2009-03-16 19:57:08.976371859 +0200
@@ -28,7 +28,7 @@
  */
 class rcmail
 {
-  static public $main_tasks = array('mail','settings','addressbook','login','logout');
+  static public $main_tasks = array('mail','settings','addressbook','faq','login','logout');
   
   static private $instance;
   
--- /dev/null	2008-11-04 20:33:38.146691408 +0200
+++ roundcubemail-0.2.1/skins/default/templates/faq.html	2009-01-21 09:22:58.000000000 +0200
@@ -0,0 +1,16 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+<title>RoundCube|Mail :: FAQ</title>
+<roundcube:include file="/includes/links.html" />
+</head>
+<body>
+
+<div id="header"><img src="/images/roundcube_logo.png" alt="RoundCube Webmail" /></div>
+
+<div style="width:400px; margin:60px auto;">
+$__page_content
+</div>
+
+</body>
+</html>
