--- roundcubemail-0.2-stable/program/steps/settings/func.inc.orig	2008-12-16 20:05:50.000000000 +0100
+++ roundcubemail-0.2-stable/program/steps/settings/func.inc	2008-12-31 00:16:53.000000000 +0100
@@ -88,7 +88,7 @@
       $select_timezone->add('(GMT -1:00) Azores, Cape Verde Islands', '-1');
       $select_timezone->add('(GMT) Western Europe, London, Lisbon, Casablanca', '0');
       $select_timezone->add('(GMT +1:00) Central European Time', '1');
-      $select_timezone->add('(GMT +2:00) EET: Kaliningrad, South Africa', '2');
+      $select_timezone->add('(GMT +2:00) EET: Tallinn, Helsinki, Kaliningrad, South Africa', '2');
       $select_timezone->add('(GMT +3:00) Baghdad, Kuwait, Riyadh, Moscow, Nairobi', '3');
       $select_timezone->add('(GMT +3:30) Tehran', '3.5');
       $select_timezone->add('(GMT +4:00) Abu Dhabi, Muscat, Baku, Tbilisi', '4');
