diff -urN roundcubemail-1.4.12-orig/config/defaults.inc.php roundcubemail-1.4.12/config/defaults.inc.php
--- roundcubemail-1.4.12-orig/config/defaults.inc.php	2021-11-12 22:39:13.000000000 +0100
+++ roundcubemail-1.4.12/config/defaults.inc.php	2021-12-29 21:41:58.988361594 +0100
@@ -435,11 +435,11 @@
 // use this folder to store log files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
 // This is used by the 'file' log driver.
-$config['log_dir'] = RCUBE_INSTALL_PATH . 'logs/';
+$config['log_dir'] = '/var/log/roundcube';
 
 // use this folder to store temp files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
-$config['temp_dir'] = RCUBE_INSTALL_PATH . 'temp/';
+$config['temp_dir'] = '/var/lib/roundcube';
 
 // expire files in temp_dir after 48 hours
 // possible units: s, m, h, d, w
