--- ./config/main.inc.php.dist.org	2006-06-17 17:42:13.000000000 +0000
+++ ./config/main.inc.php.dist	2006-06-17 17:43:35.000000000 +0000
@@ -30,10 +30,10 @@
 
 // use this folder to store log files (must be writeable for apache user)
 // This is used by the 'file' log driver.
-$rcmail_config['log_dir'] = 'logs/';
+$rcmail_config['log_dir'] = '/var/log/roundcube';
 
 // use this folder to store temp files (must be writeable for apache user)
-$rcmail_config['temp_dir'] = 'temp/';
+$rcmail_config['temp_dir'] = '/var/lib/roundcube';
 
 // enable caching of messages and mailbox data in the local database.
 // this is recommended if the IMAP server does not run on the same machine
