--- roundcubemail-0.3.1/index.php.orig	2009-10-31 14:20:02.000000000 +0100
+++ roundcubemail-0.3.1/index.php	2009-11-29 18:12:59.000000000 +0100
@@ -144,6 +144,11 @@
     $OUTPUT->show_message('sessionerror', 'error');
     $RCMAIL->kill_session();
   }
+} 
+
+// faq page
+else if ($RCMAIL->task=='faq') {
+       $OUTPUT->send($RCMAIL->task);
 }
 
 // don't check for valid request tokens in these actions
--- roundcubemail-0.4/program/include/rcmail.php~	2010-08-06 17:59:34.000000000 +0300
+++ roundcubemail-0.4/program/include/rcmail.php	2010-08-23 13:33:22.411426529 +0300
@@ -28,7 +28,7 @@
  */
 class rcmail
 {
-  static public $main_tasks = array('mail','settings','addressbook','login','logout','utils','dummy');
+  static public $main_tasks = array('mail','settings','addressbook','faq','login','logout','utils','dummy');
 
   static private $instance;
 
--- /dev/null	2008-11-04 20:33:38.146691408 +0200
+++ roundcubemail-0.2.1/skins/default/templates/faq.html	2009-01-21 09:22:58.000000000 +0200
@@ -0,0 +1,16 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+<title>RoundCube|Mail :: FAQ</title>
+<roundcube:include file="/includes/links.html" />
+</head>
+<body>
+
+<div id="header"><img src="/images/roundcube_logo.png" alt="RoundCube Webmail" /></div>
+
+<div style="width:400px; margin:60px auto;">
+$__page_content
+</div>
+
+</body>
+</html>
