--- roundcubemail-1.0.8-dep/config/defaults.inc.php~	2015-12-23 10:52:43.000000000 +0200
+++ roundcubemail-1.0.8-dep/config/defaults.inc.php	2016-01-18 15:21:12.144699738 +0200
@@ -309,10 +309,10 @@
 
 // use this folder to store log files (must be writeable for apache user)
 // This is used by the 'file' log driver.
-$config['log_dir'] = RCUBE_INSTALL_PATH . 'logs/';
+$config['log_dir'] = '/var/log/roundcube';
 
 // use this folder to store temp files (must be writeable for apache user)
-$config['temp_dir'] = RCUBE_INSTALL_PATH . 'temp/';
+$config['temp_dir'] = '/var/lib/roundcube';
 
 // expire files in temp_dir after 48 hours
 // possible units: s, m, h, d, w
