--- roundcubemail-0.4/program/include/main.inc~	2010-09-16 13:01:45.350103817 +0300
+++ roundcubemail-0.4/program/include/main.inc	2010-09-16 13:02:12.636921259 +0300
@@ -29,9 +29,6 @@
 require_once('lib/utf7.inc');
 require_once('include/rcube_shared.inc');
 
-// fallback if not PHP modules are available
-@include_once('lib/utf8.class.php');
-
 // define constannts for input reading
 define('RCUBE_INPUT_GET', 0x0101);
 define('RCUBE_INPUT_POST', 0x0102);
